client/shared/src/main/scala/org/http4s/client/Client.scala

```diff
@@ -289,16 +291,54 @@ object Client {
       go(source).stream
     }
 
-    def runOrDispose(req: Request[F]): F[Resource[F, Response[F]]] =
-      Ref[F].of(false).map { disposed =>
-        val req0 = addHostHeaderIfUriIsAbsolute(req.pipeBodyThrough(until(disposed)))
+    def processResponse(
+        response: Response[F],
+        disposed: Ref[F, Boolean],
+    ): Resource[F, Response[F]] =
+      for {
+        channel <- Resource.eval(Channel.synchronous[F, Chunk[Byte]])
+
+        _ <- response.body.chunks
+          .through(channel.sendAll)
+          .compile
+          .drain
+          .uncancelable
+          .background
```

Comment:
Will this make sense if `disposed` is `true`?

Reply:
I don't think it matters, because the resource is already closed when `disposed` is `true`. The user can only try to re-read the body, that contains this check.

Reply:
Hm, actually I just realized there may be a problem here 🤔 😅 

What happens if this step of the resource completes, but then it is canceled right here, before creating the `Response`? It will try to clean up by stopping this background fiber, but it can't, because it's `uncancelable`, and it will never complete naturally because the `onFinalize` to `drain` it hasn't been registered yet 🤔 

I think we have to keep the `uncancelable`, because we learned that starting this `background` will run some tasks eagerly. So as soon as we start it I think we must finish it.

Maybe instead of using `background` we do a custom `Resource.make(...)(...)` that starts the fiber and makes sure to drain the channel in the finalizer.

Reply:
If the main point is to drain the channel, then I think we can ignore the background process and add the finalizer to the release stage of the channel.